packages/@aws-cdk/core/lib/validation/private/report.ts

```diff
@@ -0,0 +1,190 @@
+    const vertical = '│';
+    function renderTraceInfo(info?: ConstructTrace, indent?: string, start: string = starter): string {
+      if (info) {
+        const indentation = indent ?? ' '.repeat(starter.length+1);
+        const result: string[] = [
+          `${start} ${info?.id} (${info?.path})`,
+          `${indentation}${vertical} Library: ${info?.library}`,
+          `${indentation}${vertical} Library Version: ${info?.libraryVersion}`,
+          `${indentation}${vertical} Location: ${info?.location}`,
+          ...info?.parent ? [renderTraceInfo(info?.parent, ' '.repeat(indentation.length+starter.length+1), indentation+starter)] : [],
+        ];
+        return result.join('\n\t');
+      }
+      return notAvailableMessage;
+    }
+    if (constructPath) {
+      const treeNode = this.tree.getTreeNode(constructPath);
+      if (treeNode) {
+        const traceInfo = this.getConstructTrace(treeNode);
+        return renderTraceInfo(traceInfo);
+      }
+    }
+    return notAvailableMessage;
+  }
+
+  public toString(reps: report.ValidationReport[]): string {
+    const json = this.toJson(reps);
+    const output = [json.title];
+
+    output.push('-'.repeat(json.title.length));
+    json.pluginReports.forEach(plugin => {
+      output.push('');
+      output.push('(Summary)');
+      output.push('');
+      output.push(table([
+        ['Status', plugin.summary.status],
+        ['Plugin', plugin.summary.pluginName],
+        ...Object.entries(plugin.summary.metadata ?? {}),
+      ]));
+
+      if (plugin.violations) {
+        output.push('');
+        output.push('(Violations)');
+      }
+
+      plugin.violations.forEach((violation) => {
+        const constructs = violation.violatingConstructs;
+        const occurrences = constructs.length;
+        const title = reset(red(bright(`${violation.ruleName} (${occurrences} occurrences)`)));
+        output.push('');
+        output.push(title);
+        output.push('');
+        output.push('  Occurrences:');
+        for (const construct of constructs) {
+          output.push('');
+          output.push(`    - Construct Path: ${construct.constructPath}`);
+          output.push(`    - Template Path: ${construct.templatePath}`);
+          output.push(`    - Creation Stack:\n\t${construct.constructStack}`);
+          output.push(`    - Resource Name: ${construct.resourceName}`);
+          if (construct.locations) {
+            output.push('    - Locations:');
+            for (const location of construct.locations) {
+              output.push(`      > ${location}`);
+            }
+          }
+        }
+        output.push('');
+        output.push(`  Recommendation: ${plugin.violations[0].recommendation}`);
+        if (plugin.violations[0].fix) {
+          output.push(`  How to fix: ${plugin.violations[0].fix}`);
+        }
+      });
+
+    });
+
+    return output.join(os.EOL);
+  }
+
+  public toJson(reps: report.ValidationReport[]): report.ValidationReportJson {
```

Comment:
Would it be easy to provide a JSON output option? Maybe via a CLI option or something.

Reply:
Yes, I'm not sure how easy it is, but we should do it.

Reply:
Done.